code/modules/clothing/glasses/glasses.dm

```diff
@@ -52,20 +52,21 @@
 	//In case someone in the future adds a non-toggle action to a child type
 	if(istype(action, /datum/action/item_action/toggle))
 		var/datum/action/item_action/toggle/toggle = action
-		toggle.toggled = !activate(user)
+		//Toggling item actions are dumb; it has to be done this way for the icon to properly update
+		toggle.set_toggle(activate(user))
```

Comment:
? What is happening here? Hitting a toggle action already toggles the button if you're actually returning a value properly.

Reply:
I wanted the icon in the button to update but funny thing, the point where the `toggled` state is checked only happens if the return value of `ui_action_whatever` was not false. Which is why it only updated half the time. This way it guarantees it is always updated when `ui_action_whatever` is called.